### ISSUE TITLE
Some question about Magic Point

### ISSUE THREAD

**Issue by 904281665** [rpautrat/SuperPoint#161]
Hi,excuse me. I am trying to reproduce your work on Pytorch. The detector loss I understand is actually designing a 0-2 matrix as the label-[B,64,H,W], where the label is 2 is the location of the feature point, and then a full 1 matrix is ​​added to the channel to form the label-[B,65,H,W ]. Finally, the label and the detector['logits'] after Softmax activation (channel direction) calculate the cross entropy loss.
But the feature point detection effect of the trained model is very poor, is it because there is a problem with my design?

**Comment by rpautrat** [comment]
Hi, I think you missed the step where the label is the argmax of the 0-2 matrix, like in [https://github.com/rpautrat/SuperPoint/blob/61bc16ff0e096340a26feb4cf80bf6b4bacee267/superpoint/models/utils.py#L61](https://github.com/rpautrat/SuperPoint/blob/61bc16ff0e096340a26feb4cf80bf6b4bacee267/superpoint/models/utils.py#L61).

**Comment by 904281665** [comment]
Thanks for your help！But I can't understand this code.Does it means that ?
![image](https://user-images.githubusercontent.com/43224350/89485066-84d6e000-d7d2-11ea-829d-c69876172a0f.png)
I reproduce this block by the code as follows:
label - [B, 65, H, W], logits - [B, 65, H, W]
np.transpose(label, (0, 2,3,1))
label += np.random.uniform(0, 0.1, label.shape)
label = np.argmax(label, axis=3)
logits = Softmax(logits, axis=1)
loss = Cross_entry(label.reshape(-1), logits.reshape(-1, 65))

**Comment by rpautrat** [comment]
Hi, yes the addition of the random.uniform is here to do what is said in the footnote. So I think you have here all the necessary steps.
Do you still have poor results on the synthetic dataset with this code?

**Comment by 904281665** [comment]
Unfortunately, the answer is yes. During training, when the loss dropped from 11 to 3.1 or so, the decline stopped.And when the learning rate is 0.001, loss=nan will appear in the later stage of training. Is it because I did not pre-train on ImageNet?

**Comment by 904281665** [comment]
I think I solved it. Before, I only added a layer of BN at the end of the model. After I added BN after each layer, the loss started to decrease! Thank you so much.

**Comment by rpautrat** [comment]
Great that you solved it! Thanks for posting the solution that made it work.
